pkg/vsphere/disk/disk_manager.go

```diff
@@ -269,7 +272,11 @@ func (m *Manager) Get(op trace.Operation, diskURI *object.DatastorePath, fst Fil
 // }
 
 // Attach attempts to attach a virtual disk
-func (m *Manager) Attach(op trace.Operation, disk *types.VirtualDisk) error {
+func (m *Manager) Attach(op trace.Operation, config *VirtualDiskConfig) error {
+	defer trace.End(trace.Begin(""))
+
+	disk := m.toSpec(config)
```

Comment:
I don't see a check for whether the disk is already attached - given this method is exposed let's make it idempotent.